File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/digitSum.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/digitSum
 *
 * @author Alex Golub
 * @since 03-May-21, 7:43 PM
 */
class digitSum {
  public static void main(String[] args) {
    System.out.println(digitSum(1729));
    System.out.println(digitSum(-1729)); // -19
  }
  
  public static int digitSum(int n) {
    return Math.abs(n) < 10 ? n : (n < 0 ? -1 : 1) * Math.abs(n) % 10 + digitSum(n / 10);
  }
}

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/digitSum.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/digitSum
 *
 * @author Alex Golub
 * @since 03-May-21, 7:50 PM
 */
function digitSum(n) {
  const neg = n < 0;
  if (Math.abs(n) < 10) {
    return n;
  }

  const result = Math.abs(n) % 10 + digitSum(Math.floor(Math.abs(n) / 10));
  return neg ? -result : result;
}

console.log(digitSum(-1729));
console.log(digitSum(-7));
